### PR TITLE
[action][git_submodule_update] documentation improvements

### PR DESCRIPTION
**Pull request by crazymanish**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

- `git_submodule_update` action documentation was slightly off/incorrect 😇

### Description
- Updated the description
- Updated the ConfigItem(s), make them consistent, fixed alignment and added verify_block. 
- Added **example_code** 

### Testing Steps
- Nothing changed in functionality.
- All existing unit tests are passing i.e in `git_submodule_update_spec.rb`

### CODE REVIEW

**Comment by crazymanish** [comment]
One match unit-test failed which is completely unrelated 😇

Screenshot 2021-04-24 at 23 47 29

File: fastlane/lib/fastlane/actions/git_submodule_update.rb [hunk 1]
```diff
@@ -13,20 +13,27 @@ def self.run(params)
...
 
       def self.available_options
         [
           FastlaneCore::ConfigItem.new(key: :recursive,
-           description: "Should the submodules be updated recursively",
-           type: Boolean,
-           default_value: false),
+                                       description: "Should the submodules be updated recursively?",
+                                       type: Boolean,
+                                       is_string: false,
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/git_submodule_update.rb:24]: `is_string` can be removed 😊 `type` overrides `is_string` and `is_string` is starting to get phased out

File: fastlane/lib/fastlane/actions/git_submodule_update.rb [hunk 2]
```diff
@@ -13,20 +13,27 @@ def self.run(params)
...
+                                         UI.user_error!("Please pass a valid value for recursive. Use one of the following: true, false") unless value.kind_of?(TrueClass) || value.kind_of?(FalseClass)
+                                       end),
           FastlaneCore::ConfigItem.new(key: :init,
-           description: "Should the submodules be initiated before update",
-           type: Boolean,
-           is_string: false,
-           default_value: false)
+                                       description: "Should the submodules be initiated before update?",
+                                       type: Boolean,
+                                       is_string: false,
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/git_submodule_update.rb:32]: Same here - `is_string` can be removed

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Some suggestions on things that can be removed 😊

**Comment by crazymanish** [comment]
Thank you for your review, applied suggestions! 😇 @joshdholtz

**Review by joshdholtz — APPROVED** [review]
This is much better! Thanks for improving this action ❤️ Really appreciate another one of your contributions!

**Pull request merged**